### ISSUE TITLE
Support for deleting the persistent file created in s3 when using Athena while using custom query. 

### ISSUE THREAD

**Issue by utkarshshukla2912** [fivetran/great_expectations#2872]
GE needs to create a temp table when custom query is passed, Athen does not support this which leads to creating an external table. We can drop this table later so that it is not picked up by meta data crawler but a persistent file is created in s3.

- Is there a way to avoid creating this persistent file in s3 ?
- If answer is no is there a way to delete after its created ?

**Comment by mconfalonieri** [comment]
I have a similar problem, as the name of the "temporary" Athena table is not stored anywhere, I have no possibility to retrieve it.
I tried to use `bigquery_temp_table` as workaround, and although it works, it insists to write the temporary table to the `default` database, even if a `schema` parameter is provided. At this point, if the table is successfully created, it tries to read it from the database provided in the `schema` parameter:

```
[2021-06-10 10:22:16,537] {credentials.py:1100} INFO - Found credentials in environment variables.
[2021-06-10 10:22:16,763] {credentials.py:1100} INFO - Found credentials in environment variables.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.9/site-packages/pyathena/util.py", line 37, in _wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyathena/cursor.py", line 108, in execute
    raise OperationalError(query_execution.state_change_reason)
pyathena.error.OperationalError: SYNTAX_ERROR: line 2:6: Table awsdatacatalog.airflow-temp.tmp_check_roughness_daily does not exist

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/marcon/src/NCS-millard/Core/services/roughness-data-validation/src/ge_roughness_dag.py", line 112, in <module>
    context.run_checkpoint("check_roughness_daily_aggregation")
  File "/usr/local/lib/python3.9/site-packages/great_expectations/data_context/data_context.py", line 3053, in run_checkpoint
    return checkpoint.run(
  File "/usr/local/lib/python3.9/site-packages/great_expectations/checkpoint/checkpoint.py", line 559, in run
    batches_to_validate = self._get_batches_to_validate(self.batches)
  File "/usr/local/lib/python3.9/site-packages/great_expectations/checkpoint/checkpoint.py", line 621, in _get_batches_to_validate
    batch = self.data_context.get_batch(batch_kwargs, suite)
  File "/usr/local/lib/python3.9/site-packages/great_expectations/data_context/data_context.py", line 1436, in get_batch
    return self._get_batch_v2(
  File "/usr/local/lib/python3.9/site-packages/great_expectations/data_context/data_context.py", line 1153, in _get_batch_v2
    return validator.get_dataset()
  File "/usr/local/lib/python3.9/site-packages/great_expectations/validator/validator.py", line 1490, in get_dataset
    return self.expectation_engine(
  File "/usr/local/lib/python3.9/site-packages/great_expectations/dataset/sqlalchemy_dataset.py", line 629, in __init__
    self.columns = self.column_reflection_fallback()
  File "/usr/local/lib/python3.9/site-packages/great_expectations/dataset/sqlalchemy_dataset.py", line 1378, in column_reflection_fallback
    col_names: list = self.engine.execute(query).keys()
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2235, in execute
    return connection.execute(statement, *multiparams, **params)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.9/site-packages/pyathena/util.py", line 37, in _wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyathena/cursor.py", line 108, in execute
    raise OperationalError(query_execution.state_change_reason)
sqlalchemy.exc.OperationalError: (pyathena.error.OperationalError) SYNTAX_ERROR: line 2:6: Table awsdatacatalog.airflow-temp.tmp_check_roughness_daily does not exist
[SQL: SELECT *
FROM "airflow-temp"."tmp_check_roughness_daily"
 LIMIT %(param_1)s]
[parameters: {'param_1': 1}]
```

I saw that there was an issue related to this, #1964, but it seems that the fix had some side effects.

**Comment by eugmandel** [comment]
@utkarshshukla2912 Thank you for creating this issue! Labeling as "help wanted".

**Comment by utkarshshukla2912** [comment]
@mconfalonieri if you have generated a batch you can identify the name of the table by using 
`batch.generated_table_name`

**Comment by mconfalonieri** [comment]
> @mconfalonieri if you have generated a batch you can identify the name of the table by using
> `batch.generated_table_name`

Thank you for your answer, in the end we wrote some custom expectations to avoid the use of temporary tables.

**Comment by utkarshshukla2912** [comment]
@eugmandel One work around to this could be creating views since they are not a physical table hence no storage required in s3. On completion of the profile/expectatios the batch can have a boolean parameter that triggers `DROP VIEW IF EXISTS QUERY` if the parameter is true.

One major caveat to this is that the aws credentials used to connect to Athena need to have createView access.

**Comment by AFineDayFor** [comment]
Howdy @utkarshshukla2912; we have a merged PR for this, it's resolved with #4103.
The fix provided is for v3, this isn't something we're prioritising in v2.
